### ISSUE TITLE
feat: Update to shorebird 3.13.0

### ISSUE THREAD

**Issue by eseidel** [shorebirdtech/shorebird#1125]
Making a bug so I have some place to put notes.

**Comment by eseidel** [comment]
Following https://github.com/shorebirdtech/shorebird/blob/main/packages/cutler/FORKING_FLUTTER.md

**Comment by eseidel** [comment]
Current Shorebird (0.13.1) revisions:
```
  flutter   03e895ee09dfbb9c18681d103f4b27671ff65429
  engine    f9578b37c4490c75cc66b650fcba51e221c78411
  dart      4b298415eec861a3e7f4249ee26d5c8ed79e9ebc
  buildroot 59ef3bc35a2aa6d27d065ef4be146d0d042ff191
```

Fork points:
```
flutter f468f3366c26a5092eb964a230ce7892fda8f2f8 (3.10.6)
engine  cdbeda788a293fa29665dc3fa3d6e63bd221cb0d (3.10.6)
dart 8d5b878d0a8a06326227cde88221eea7e15ee0c4 (3.0.6)
buildroot f24f62fa5381c0e415b6ca2000600fc0600c11c8
```

New Fork Points (3.13.0)
```
  flutter. efbf63d9c66b9f6ec30e9ad4611189aa80003d31 (3.13.0)
  engine    1ac611c64eadbd93c5f5aba5494b8fc3b35ee952 (3.13.0)
  dart      1b9c3b35ced9cb3a22f157f827c7ef79faa7b123 (3.1.0)
  buildroot 6e71c38443c0bf9d8954c87bf69bb4e019f44f94
```

Rebase commands:
```
git -C flutter rebase --onto 3.13.0 3.10.6 origin/shorebird/dev
git -C engine/src/flutter rebase --onto 3.13.0 3.10.6 origin/shorebird/dev
git -C engine/src/third_party/dart rebase --onto 3.1.0 3.0.6 origin/shorebird/dev
git -C engine/src rebase --onto 6e71c38443c0bf9d8954c87bf69bb4e019f44f94 f24f62fa5381c0e415b6ca2000600fc0600c11c8 origin/shorebird/dev
```

**Comment by eseidel** [comment]
Update: I have our changes successfully rebased on Dart 3.1.0, just crashing when run.  I've implemented what I believe to be the fix, but it's not working yet.  Will revisit Monday morning with more brain power.  Sorry for the delay on 3.13.0, we were in the middle of a big release and weren't ready to start the update work right when it came out.

**Comment by eseidel** [comment]
Kevin and I have worked all day on Dart 3.1 updates but are not done (either with the work or our days), but I don't see us getting Flutter 3.13.0 out today, sorry.  Still hopeful for tomorrow.  This was an extra tricky change (Dart 3.0 to 3.1) in terms of breaking our custom Dart modifications.

**Comment by eseidel** [comment]
Update!  We found what we believe to be the last crasher!  https://github.com/dart-lang/sdk/issues/53303. Now we just have to finish integrating the new Dart into Flutter Engine and verify that it works as expected and then release.  If all goes well we should have 3.13.0 out tomorrow before noon.  🤞

**Comment by eseidel** [comment]
Smoke tests:

- [x] iOS release
    - [x] Windows Machine
    - [x] MacOS Machine
- [x] iOS patch
    - [x] Windows Machine
    - [x] MacOS Machine
- [x] iOS preview
    - [x] Windows Machine
    - [x] MacOS Machine

- [x] Android release
    - [x] Windows Machine
    - [x] MacOS Machine
- [x] Android patch
    - [x] Windows Machine
    - [x] MacOS Machine
- [x] Android preview
    - [x] Windows Machine
    - [x] MacOS Machine

- [x] iOS Add to App release
    - [x] Windows Machine
    - [x] MacOS Machine
- [x] iOS Add to App patch
    - [x] Windows Machine
    - [x] MacOS Machine
- [x] iOS Add to App preview
    - [x] Windows Machine
    - [x] MacOS Machine

- [x] Android Add to App release
    - [x] Windows Machine
    - [x] MacOS Machine
- [x] Android Add to App patch
    - [x] Windows Machine
    - [x] MacOS Machine

- [x] iOS Flavors release
    - [x] Windows Machine
    - [x] MacOS Machine
- [x] iOS Flavors patch
    - [x] Windows Machine
    - [x] MacOS Machine
- [x] iOS Flavors preview
    - [x] Windows Machine
    - [x] MacOS Machine

- [ ] Android Flavors release
    - [ ] Windows Machine
    - [x] MacOS Machine
- [ ] Android Flavors patch
    - [ ] Windows Machine
    - [x] MacOS Machine
- [ ] Android Flavors preview
    - [ ] Windows Machine
    - [x] MacOS Machine

**Comment by eseidel** [comment]
Crash on iOS during preview of `samples/flavors`
```
eseidel@erics-mbp flavors % shorebird preview
✓ Fetching apps (0.2s)
Which app would you like to preview? flavors (stable)
✓ Fetching releases (98ms)
Which release would you like to preview? 1.0.0+2
Which platform would you like to preview? ios
✓ Started app (7.9s)
2023-08-22 16:02:05.468303-0700 Runner[59161:6412855] SANITY CHECK: Running precompiled code.
2023-08-22 16:02:05.468485-0700 Runner[59161:6412855] Using App.framework from /private/var/containers/Bundle/Application/9DA0F792-94FA-4AF4-A5F5-53A4CFA4474D/ios_1.0.0+2.app/Frameworks/App.framework
2023-08-22 16:02:05.468706-0700 Runner[59161:6412855] ASSET PATH /private/var/containers/Bundle/Application/9DA0F792-94FA-4AF4-A5F5-53A4CFA4474D/ios_1.0.0+2.app/Frameworks/App.framework/flutter_assets
2023-08-22 16:02:05.473944-0700 Runner[59161:6412855] [VERBOSE0:shorebird.cc(102)] Shorebird updater: no active patch.
2023-08-22 16:02:05.473957-0700 Runner[59161:6412855] [VERBOSE0:shorebird.cc(106)] Starting Shorebird update
2023-08-22 16:02:05.474222-0700 Runner[59161:6412855] [VERBOSE0:FlutterViewController.mm(235)] CPU::Id(): simarm64
2023-08-22 16:02:05.967113-0700 Runner[59161:6412855] [VERBOSE-2:FlutterDarwinContextMetalImpeller.mm(37)] Using the Impeller rendering backend.
[00:00:02.409] (16c337000) ERROR  Update failed: create_dir_all
Caused by:
    Operation not permitted (os error 1)
[00:00:02.409] (16c337000) INFO   Update thread finished with status: Update had error
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGKILL
    frame #0: 0x00000001f3d1eca4 libsystem_kernel.dylib`mach_msg2_trap + 8
libsystem_kernel.dylib`mach_msg2_trap:
->  0x1f3d1eca4 <+8>: ret    
libsystem_kernel.dylib`macx_swapon:
    0x1f3d1eca8 <+0>: mov    x16, #-0x30
    0x1f3d1ecac <+4>: svc    #0x80
    0x1f3d1ecb0 <+8>: ret    
Target 0: (Runner) stopped.
```

**Comment by eseidel** [comment]
Except I'm not sure I even patched 2.

```
eseidel@erics-mbp flavors % shorebird patch ios-alpha
[WARN] iOS support is alpha. See https://docs.shorebird.dev/status#ios-alpha for more information.
✓ Fetching apps (0.4s)
✓ Building release (38.8s)
✓ Detected release version 1.0.0+2 (21ms)
✓ Fetching releases (0.1s)
✗ No artifact found for architecture ipa in release 5525 (92ms)
```

**Comment by eseidel** [comment]
I'm not sure what was causing that before, but I was able to release and patch and preview just now.

**Comment by eseidel** [comment]
I've chosen not to test Windows Android Flavors.  I believe we're ready for release @bryanoltman

**Comment by rikousik** [comment]
Hi,
I know you guys are working hard and might be busy with some other critical stuffs, but this issue is also important for a lot of users. For example: there are some plugins that are updated to a version which only supports 3.13 and does not compile with shorebird. The only way to overcome this problem right now is to downgrade flutter to an older version in order to debug with flutter and release with shorebird successfully.

**Comment by eseidel** [comment]
Sorry I should have closed this.  We released this update this morning as Shorebird 0.14.0.  Let us know if you're still having issues!

**Comment by bryanoltman** [comment]
@ambanikousik you can join our [Discord server](https://discord.gg/shorebird) to stay up-to-date with our latest releases
